mitiq/mitiq_pyquil/tests/test_zne_mitiq_pyquil.py

```diff
@@ -15,64 +15,81 @@
 
 """Tests for zne.py with PyQuil backend."""
 import numpy as np
-from mitiq._typing import QPROGRAM
-from mitiq.zne.inference import RichardsonFactory
+
+import pyquil
+
+from mitiq import QPROGRAM
 from mitiq.zne import (
+    inference,
+    scaling,
     execute_with_zne,
     mitigate_executor,
     zne_decorator,
 )
 from mitiq.mitiq_pyquil.pyquil_utils import (
-    random_identity_circuit,
-    measure,
-    run_program,
-    scale_noise,
+    generate_qcs_executor,
+    ground_state_expectation,
 )
+from mitiq.mitiq_pyquil.conversions import to_pyquil
+from mitiq.benchmarks.randomized_benchmarking import rb_circuits
 
 TEST_DEPTH = 30
 
 
-def basic_executor(qp: QPROGRAM, shots: int = 500) -> float:
-    return run_program(qp, shots)
+def random_one_qubit_identity_circuit(num_cliffords: int) -> pyquil.Program:
+    """Returns a single-qubit identity circuit.
+
+    Args:
+        num_cliffords (int): Number of cliffords used to generate the circuit.
+
+    Returns:
+        circuit: Quantum circuit as a :class:`pyquil.Program` object.
+    """
+    return to_pyquil(
+        rb_circuits(n_qubits=1, num_cliffords=[num_cliffords], trials=1)[0]
+    )
+
+
+QVM = pyquil.get_qc("1q-qvm")
+QVM.qam.random_seed = 1337
+noiseless_executor = generate_qcs_executor(
+    qc=pyquil.get_qc("1q-qvm"),
+    expectation_fn=ground_state_expectation,
+    shots=1_000,
+)
```

Comment:
All tests seem based on a `noiseless_executor` so we actually never test that mitigation works.
This may be temporarily ok, but we could add a TODO or make a new issue about adding simulated native noise on pyquil tests.

Reply:
The current framework is that we test errors are mitigated in the zne module, while the `mitiq_qiskit` and `mitiq_pyquil` modules only test that the pipeline works for these packages. In my opinion this is sufficient, but please open a separate issue if you see otherwise.